### PR TITLE
change reservation amount & adjust while loop

### PR DESCRIPTION
**Pull request by JensOgorek**
This contains the change for the while loop and the reservation amount. The changes were added after creation of the parser with the .sh script.

this is for #3

### CODE REVIEW

File: src/6a-generate.c [hunk 1]
```diff
@@ -465,7 +465,7 @@ void generate(struct generator *gen)
     output_line(out, "static bool grow_tree(struct %%prefix_tree *tree, size_t size)");
     output_line(out, "{");
     output_line(out, "    size_t n = tree->parse_tree_size;");
-    output_line(out, "    while (n < size || n < 4096)");
+    output_line(out, "    while (n < size)"); // Issue: Zauberzeug/lizard#24 - while (n < size || n < 4096) -> while (n < size)
```
Comment (falkoschindler) [inline src/6a-generate.c:468]:
```suggestion
    output_line(out, "    while (n < size)"); // https://github.com/zauberzeug/lizard/issues/23 - remove limit of 4096
```

File: diff_list.md [hunk 2]
```diff
@@ -5,3 +5,7 @@ Changes in this fork
 
 owl_example_parser.sh:
     - Add an example script to generate the parser from the grammar
+
+src/6a-generate.c:
+    - Increase RESERVATION_AMOUNT to 11 (Zauberzeug issue #7)
+    - Change the while loop in the read_tree function to while (n < size) (Zauberzeug issue #24)
```
Comment (falkoschindler) [inline diff_list.md:11]:
```suggestion
src/6a-generate.c:
    - Increase `RESERVATION_AMOUNT` to 11 (https://github.com/zauberzeug/field_friend/issues/7)
    - Remove limit of 4096 from while loop in the `read_tree()` (https://github.com/zauberzeug/lizard/issues/23)
```

File: src/6a-generate.c [hunk 3]
```diff
@@ -443,7 +443,7 @@ void generate(struct generator *gen)
 
     // Code for reading and writing packed parse trees.
     output_line(out, "// Reserve 10 bytes for each entry (the maximum encoded size of a 64-bit value).");
-    output_line(out, "#define RESERVATION_AMOUNT 10");
+    output_line(out, "#define RESERVATION_AMOUNT 11"); // Issue: Zauberzeug/field_friend#7 - increase from 10 to 11
```
Comment (falkoschindler) [inline src/6a-generate.c:446]:
```suggestion
    output_line(out, "#define RESERVATION_AMOUNT 11"); // https://github.com/zauberzeug/field_friend/issues/7 - increase from 10 to 11
```

**Review by falkoschindler — CHANGES_REQUESTED** [review]
Let's use full GitHub links so that one can reach them more easily.
And the while loop has been changed in issue 23, not 24.

File: src/6a-generate.c (hunk 1 above)
Comment (JensOgorek) [reply src/6a-generate.c]: updated

File: diff_list.md (hunk 2 above)
Comment (JensOgorek) [reply diff_list.md]: updated

**Review by falkoschindler — APPROVED** [review]

**Pull request merged**